src/data/simple_dmatrix.cc

```diff
@@ -41,7 +41,7 @@ BatchSet<SparsePage> SimpleDMatrix::GetRowBatches() {
 BatchSet<CSCPage> SimpleDMatrix::GetColumnBatches() {
   // column page doesn't exist, generate it
   if (!column_page_) {
-    auto page = dynamic_cast<SimpleCSRSource*>(source_.get())->page_;
+    auto const& page = dynamic_cast<SimpleCSRSource*>(source_.get())->page_;
```

Comment:
Isnt this a pointer, so there is no copy?

Reply:
No, it's not a pointer.  See the tailing `page_`

Reply:
That is quite bad that this could happen. Good job finding it.